src/ws/cockpitwebservice.c

```diff
@@ -1134,6 +1134,7 @@ on_web_socket_open (WebSocketConnection *connection,
   json_object_set_string_member (object, "channel-seed", socket->id);
   json_object_set_string_member (object, "host", "localhost");
   json_object_set_string_member (object, "csrf-token", cockpit_creds_get_csrf_token (self->creds));
+  json_object_set_object_member (object, "bridge-init", json_object_ref (cockpit_web_service_get_init (self)));
```

Comment:
So AFAICS,  the bridge code is sort of right, but this is the wrong integration point -- this only runs once at session startup (for the primary session bridge), but not with remote connections. I think we entirely discard the peer's init message in src/cockpit/peer.py `def start()`. Instead the bridge would need to keep a host → init message map and we turn `bridge-init` from a static init message field into a more dynamic bridge call. In the worst case this could be similar to how we do `cockpit.user()`, via a bridge D-Bus call? (I can't think of a better way, we don't get a new `cockpit.transport` for these, do we?

Reply:
An internal endpoint could do a similar thing and it would be host-friendly.  It would also be much more annoying to use....